src/services/email.integration-test.ts

```diff
@@ -0,0 +1,96 @@
+import { expect, getMails, mocks, resolveOnEvent } from "../utils/tests-setup";
+import {
+  ConfigurableSendDeps,
+  EmailMessage,
+  TRANSPORT_NAME,
+  composeMessage,
+  configurableInitialize,
+  configurableSend,
+  notify,
+} from "./email";
+
+describe("Email service", () => {
+  let emitter: EventEmitter;
+
+  beforeEach(() => (emitter = new EventEmitter()));
+  describe("initialize", () => {
+    const fn = configurableInitialize;
+
+    it("should emit an event when it is properly initialized", async () => {
+      const promise = resolveOnEvent(BotEventNames.TRANSPORT_READY, emitter);
+      fn({
+        emitter,
+        env,
+        createTransportFn: nodemailer.createTransport.bind(nodemailer),
+      });
+      const event = (await promise)[0] as TransportReadyPayload;
+      expect(event.name).to.equal(TRANSPORT_NAME);
+    });
+
+    describe("should emit an event when a configuration environment variable missing", () => {
+      const requiredFields: Array<keyof Env> = [
+        "SMTP_HOST",
+        "SMTP_PORT",
+        "SMTP_USER",
+        "SMTP_PASSWORD",
+        "SMTP_FROM",
+        "SMTP_TO",
+      ];
+      requiredFields.forEach((key) => {
+        it(`should emit a missing event when ${String(key)} is missing`, async () => {
+          const promise = resolveOnEvent(BotEventNames.TRANSPORT_CONFIGURATION_MISSING, emitter);
+          const fakeEnv = { ...env, [key]: undefined };
+          fn({
+            emitter,
+            createTransportFn: nodemailer.createTransport.bind(nodemailer),
+            env: fakeEnv,
+          });
+          const event = (await promise)[0] as TransportConfigurationMissingPayload;
+          expect(event.name).to.equal(TRANSPORT_NAME);
+          expect(event.missing).to.have.length(1);
+          expect(event.missing[0]).to.equal(key);
+        });
+      });
+    });
+  });
+
+  describe("notify", () => {
+    const fn = notify;
+
+    it("should notify when the integration is configured", async () => {
+      const previous = await getMails();
+
+      await fn(mocks.proposalMock);
+
+      const current = await getMails();
+      expect(current).to.have.lengthOf(previous.length + 1);
+
+      const email = current.pop()!;
+      expect(email.from[0].address).to.equal(env.SMTP_FROM);
+      expect(email.to[0].address).to.equal(env.SMTP_TO);
+
+      const expectedContent = composeMessage(mocks.proposalMock);
+      expect(email.subject).to.equal(expectedContent.subject);
+      expect(email.text).to.equal(expectedContent.text);
+    });
+  });
+
+  describe("send", () => {
+    const fn = configurableSend;
+
+    it("should avoid sending without raising errors when the client is not configured", async () => {
+      const depsMock = {
+        sendFn: undefined,
+        from: "test@test.com",
+        to: "test@test.com",
+        message: {} as EmailMessage,
+      } as ConfigurableSendDeps;
+
+      expect(fn(depsMock)).to.be.fulfilled;
```

Comment:
I would expect a .toThrow given the description of test.

Reply:
Hehe, this one is tricky so I can understand your concern.

Basically, the application has a series of notification transports (slack, email, telegram) which should send the notifications when they are properly configured (tokens, credentials, whatever) or do nothing otherwise.

This test checks that when the transport is not configured, it does nothing. Sadly, I can not check that sendFn has not been called because I can not spy it: it is `undefined`.

Reply:
I think there are multiple was to interpret

> should avoid sending without raising errors when the client is not configured

1. Should raise errors if client is not configured - should avoid **sending without raising errors**

2. Should not raise errors - should avoid sending **without raising errors**

I interpreted it as 1, but from your reply if looks like maybe you meant 2?